### PR TITLE
Add tests for devices service

### PR DESCRIPTION
**Pull request by TsvetanMilanov**
Add unit tests for the devices-service.

### CODE REVIEW

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

File: test/unit-tests/mobile/devices-service.ts [hunk 1]
```diff
@@ -932,4 +967,520 @@ describe("devicesService", () => {
 			assert.deepEqual(devicesService.getDeviceByDeviceOption(), androidDevice);
 		});
 	});
+
+	describe("isCompanionAppInstalledOnAllDevices", () => {
+		let $companionAppService: ICompanionAppsService;
```
Comment (rosen-vladimirov) [inline test/unit-tests/mobile/devices-service.ts]: companionAppsService is the name registered in injector ;)

File: test/unit-tests/mobile/devices-service.ts [hunk 2]
```diff
@@ -932,4 +967,520 @@ describe("devicesService", () => {
...
+		let deviceIdentifiers = [iOSDevice.deviceInfo.identifier, androidDevice.deviceInfo.identifier];
+
+		beforeEach(() => {
+			$companionAppService = testInjector.resolve("companionAppsService");
+			$companionAppService.getCompanionAppIdentifier = (framework: string, platform: string): string => "companion-app";
+
+			(<any>devicesService).getDeviceByIdentifier = (identifier: string) => {
```
Comment (rosen-vladimirov) [inline test/unit-tests/mobile/devices-service.ts]: why do you need the cast?

File: test/unit-tests/mobile/devices-service.ts [hunk 3]
```diff
@@ -932,4 +967,520 @@ describe("devicesService", () => {
...
+		});
+
+		it("should return correct result when all of the devices have the companion app installed.", () => {
+			let expectedResult = [true, true];
+			mockIsAppInstalled([iOSDevice, androidDevice], expectedResult);
+
+			let result = _.map(devicesService.isCompanionAppInstalledOnDevices(deviceIdentifiers,
+				constants.TARGET_FRAMEWORK_IDENTIFIERS.Cordova), (future: IFuture<IAppInstalledInfo>) => future.wait().isInstalled);
+
+			assert.sameMembers(result, expectedResult);
```
Comment (rosen-vladimirov) [inline test/unit-tests/mobile/devices-service.ts]: isn't `deepEqual` applicable here (and for the rest of the tests). The order of the results is important, but `sameMembers` does not check the order of elements.

File: test/unit-tests/mobile/devices-service.ts [hunk 4]
```diff
@@ -115,6 +115,41 @@ function createTestInjector(): IInjector {
...
+function throwErrorFuture(): IFuture<void> {
+	return (() => {
+		throw new Error("error");
+	}).future<void>()();
+}
+
+let intervalId = 1;
+let nodeJsTimer = {
+	ref: () => { /* no implementation required */ },
+	unref: () => { return intervalId++; }
```
Comment (rosen-vladimirov) [inline test/unit-tests/mobile/devices-service.ts]:
```TypeScript
unref: () => intervalId++
```

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by rosen-vladimirov** [comment]
:+1:

**Pull request merged**